src/openrct2/paint/map_element/scenery.c

```diff
@@ -214,27 +214,25 @@ void scenery_paint(uint8 direction, sint32 height, rct_map_element* mapElement)
                     image_id = (image_id & 0x7FFFF) | dword_F64EB0;
                 }
                 sub_98199C(image_id, x_offset, y_offset, boxlength.x, boxlength.y, boxlength.z - 1, height, boxoffset.x, boxoffset.y, boxoffset.z, rotation);
-            } else {
-                if (entry->small_scenery.flags & SMALL_SCENERY_FLAG16) {
-                    // nothing
-                }
-                sint32 esi = gCurrentTicks;
-                if (!(entry->small_scenery.flags & SMALL_SCENERY_FLAG22)) {
+            }
+            else if (entry->small_scenery.flags & SMALL_SCENERY_FLAG_HAS_FRAME_OFFSETS)
```

Comment:
Yes that was on purpose. The previous logic didn't make sense. In theory the old logic could access an invalid pointer. It doesn't actually make a difference as only a corrupted . dat would hit this code wrong.